### DISCUSSION TITLE
UserWarning about regular expression - safe to ignore?

### DISCUSSION THREAD

**Discussion by jlounds** [blaylockbk/Herbie#379] [Q&A] [answered]
Greetings!

I am trying to extract multiple values at once, e.g.

`ds = H.xarray('(:RH:2)|(:TMP:2)')`

Is it safe to ignore the following message?

`herbie/core.py:823: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.`

Thank you!

Jeremy

**Comment by williamhobbs** [comment, marked as answer]
It seems to work either way, but I think this will get rid of the warning for you:

`ds = H.xarray('(?:RH:2)|(?:TMP:2)')`

You could also write it as:

`ds = H.xarray('(?:TMP|RH):2 m')`

(idea from https://stackoverflow.com/a/71374335/27574852)

**Reply by blaylockbk** [reply]
Correct, the `(?:pattern)` syntax is a "non-capturing group" meaning we are just trying to match the string. 

> A non-capturing group in regular expressions is used to group a subpattern without storing the matched text.

**Reply by jlounds** [reply]
Thank you both.
